src/codewhisperer/util/supplementalContext/codeParsingUtil.ts

```diff
@@ -19,7 +19,7 @@ export const utgLanguageConfigs: Record<string, utgLanguageConfig> = {
     // Java regexes are not working efficiently for class or function extraction
     java: {
         extension: '.java',
-        testFilenamePattern: /(?:Test([^/\\]+)\.java|([^/\\]+)Test\.java)$/,
+        testFilenamePattern: /(?:Test([^/\\]+)\.java|([^/\\]+)Test\.java|([^/\\]+)Tests\.java)$/,
```

Comment:
I still prefer a list of smaller regex like https://github.com/aws/aws-toolkit-jetbrains/blob/main/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/JavaCodeWhispererFileCrawler.kt#L25, which is easier to work with. But I will take this as follow up and don't want to bring too much changes at one shot. Any suggestion on which one is better? @justinmk3 @andrewyuq

Reply:
For lists of "or" (`|`) cases, a list of regexes is usually a good idea. Especially if the list will continue to grow.

Reply:
yea agree, it's quite error prone when working with this giant regex. I will take refactoring as a follow up.

Reply:
TODO comment?
and it looks like these two can be combined?
`([^/\\]+)Test\.java|([^/\\]+)Tests\.java`

Do we want to match aggressively or preservably? If it's the former do want to include any filename that contains the case-insensitive string `test`?

Reply:
I prefer not combining these 2 regex, I think it's much error prone. The final format should be more similar to JB's implementation which use a list of small regex, it's also for readability.

Reply:
I think case sensitive is fine, let's do it case sensitive for now. We can update it if needed.